src/utils/builder.ts

```diff
@@ -0,0 +1,297 @@
+import { Pathway, Criteria, State, Transition, Action, GuidanceState } from 'pathways-model';
+import shortid from 'shortid';
+import { MedicationRequest, ServiceRequest } from 'fhir-objects';
+
+export function createNewPathway(name: string, description?: string): Pathway {
+  return {
+    name: name,
+    description: description ?? '',
+    library: '',
+    criteria: [],
+    states: {
+      Start: {
+        key: 'Start',
+        label: 'Start',
+        transitions: []
+      }
+    }
+  };
+}
+
+export function exportPathway(pathway: Pathway): string {
+  // Strip id from each criteria
+  pathway.criteria.forEach((criteria: Criteria) => delete criteria.id);
+
+  // Strip key from each state
+  Object.keys(pathway.states).forEach((stateName: string) => {
+    const state = pathway.states[stateName];
+    delete state.key;
+
+    // Strip id from each state.transition
+    state.transitions.forEach((transition: Transition) => delete transition.id);
+
+    // Strip id from each state.action
+    if ('action' in state) {
+      (state as GuidanceState).action.forEach((action: Action) => delete action.id);
+    }
+  });
+
+  return JSON.stringify(pathway, undefined, 2);
+}
+
+export function addLibrary(pathway: Pathway, library: string): void {
+  pathway.library = library;
+}
+
+// TODO: possibly add more criteria methods
```

Comment:
We need Criteria Source to be added to the model

Reply:
From our discussion I don't think is required anymore. The Criteria source is only used to determine what the CQL is. We then use the setCQL method to set the actual CQL. Our model currently only supports libraries at the pathways level and there is a method to set that.